instrumentation/opentelemetry/internal/metrics/linux_metrics.go

```diff
@@ -0,0 +1,95 @@
+//go:build linux
+
+package metrics
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/tklauser/go-sysconf"
+)
+
+const procStatArrayLength = 52
+
+var (
+	clkTck   = getClockTicks()
+	pageSize = float64(os.Getpagesize())
+)
+
+type processStats struct {
+	utime  float64
+	stime  float64
+	cutime float64
+	cstime float64
+	rss    float64
+}
+
+type linuxMetrics struct {
+	memory          float64
+	cpuSecondsTotal float64
+}
+
+func newSystemMetrics() (systemMetrics, error) {
+	lm := &linuxMetrics{}
+	stats, err := lm.processStatsFromPid(os.Getpid())
+	if err != nil {
+		return nil, err
+	}
+	lm.memory = stats.rss * pageSize
+	lm.cpuSecondsTotal = (stats.stime + stats.utime + stats.cstime + stats.cutime) / clkTck
+	return lm, nil
+}
+
+func (lm *linuxMetrics) getMemory() float64 {
+	return lm.memory
+}
+
+func (lm *linuxMetrics) getCPU() float64 {
+	return lm.cpuSecondsTotal
+}
+
+func (lm *linuxMetrics) processStatsFromPid(pid int) (*processStats, error) {
```

Comment:
no tests for this function? For testing this
1. either file should become a parameter to the function and then you can create custom files to test.
2. can pass an io.Reader to the function to read file. (*os.File) implements an io.Reader. During test you can use a mock version.